### PR TITLE
changed confusing username/password config attributes to standard AWS id/key names

### PR DESCRIPTION
**Pull request by maxcoulombe**
# Overview
The plugin config expects a username & password to connect to ElastiCache but the field names caused confusion with users. They expect the username/password to refer to a Redis user, but ElastiCache users are managed with the AWS API and not by connecting to the underlying Redis instance in the ElastiCache cluster.

Access Key ID & Secret Access Key make more sense in this context.

Upstream doc in Vault will be updated accordingly.

Other minor changes:
* Using $GOROOT/vault-plugins for the local bootstrap binary location as this is what other projects use
* Fix Terraform warning with AWS current region
* Added the possibility of configuring the plugin with a session token
* Changed the Dependabot config to tag the whole team instead of a single person when opening PRs

# Contributor Checklist
- [-] Add relevant docs to upstream Vault repository, or sufficient reasoning why docs won’t be added yet - Upstream doc change upcoming, will update when doc PR is ready
- [x] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
```
max:~/projects/vault-plugin-database-redis-elasticache$ make testacc
ACC_TEST_ENABLED=1 CGO_ENABLED=0 go test ./...  -timeout=20m
ok  	github.com/hashicorp/vault-plugin-database-redis-elasticache	1.543s
?   	github.com/hashicorp/vault-plugin-database-redis-elasticache/cmd/vault-plugin-database-redis-elasticache	[no test files]
```

- [x] Backwards compatible
We kept the deprecated username/password fields so the change is backward compatible

### CODE REVIEW

File: bootstrap/configure.sh [hunk 1]
```diff
@@ -3,8 +3,8 @@ PLUGIN_NAME=$2
 
 TEST_ELASTICACHE_URL=$3
 TEST_ELASTICACHE_REGION=$4
-TEST_ELASTICACHE_USERNAME=$5
-TEST_ELASTICACHE_PASSWORD=$6
+$TEST_ELASTICACHE_ACCESS_KEY_ID=$5
+$TEST_ELASTICACHE_SECRET_ACCESS_KEY=$6
```
Comment (fairclothjm) [inline bootstrap/configure.sh:7]:
```suggestion
TEST_ELASTICACHE_ACCESS_KEY_ID=$5
TEST_ELASTICACHE_SECRET_ACCESS_KEY=$6
```

File: redis_elasticache_client.go [hunk 2]
```diff
@@ -23,10 +23,13 @@ type redisElastiCacheDB struct {
...
-	Password string `mapstructure:"password,omitempty"`
-	Url      string `mapstructure:"url,omitempty"`
-	Region   string `mapstructure:"region,omitempty"`
+	AccessKeyID     string `mapstructure:"access_key_id,omitempty"`
+	SecretAccessKey string `mapstructure:"secret_access_key,omitempty"`
+	SessionToken    string `mapstructure:"session_token,omitempty"`
+	Username        string `mapstructure:"username,omitempty"` // @Deprecated, use AccessKeyID instead
+	Password        string `mapstructure:"password,omitempty"` // @Deprecated, use SecretAccessKey instead
+	Url             string `mapstructure:"url,omitempty"`
+	Region          string `mapstructure:"region,omitempty"`
```
Comment (fairclothjm) [inline redis_elasticache_client.go:32]: nit: suggest to move the deprecated fields apart from the rest

```suggestion
	AccessKeyID     string `mapstructure:"access_key_id,omitempty"`
	SecretAccessKey string `mapstructure:"secret_access_key,omitempty"`
	SessionToken    string `mapstructure:"session_token,omitempty"`
	Url             string `mapstructure:"url,omitempty"`
	Region          string `mapstructure:"region,omitempty"`

	Username string `mapstructure:"username,omitempty"` // @Deprecated, use AccessKeyID instead
	Password string `mapstructure:"password,omitempty"` // @Deprecated, use SecretAccessKey instead
```

File: redis_elasticache_client.go [hunk 3]
```diff
@@ -36,7 +39,17 @@ func (r *redisElastiCacheDB) Initialize(_ context.Context, req dbplugin.Initiali
...
+	accessKey := r.config.AccessKeyID
+	if accessKey == "" && r.config.Username != "" {
+		accessKey = r.config.Username
+	}
+	secretKey := r.config.SecretAccessKey
+	if secretKey == "" && r.config.SecretAccessKey != "" {
+		secretKey = r.config.SecretAccessKey
+	}
+
+	creds, err := awsutil.RetrieveCreds(accessKey, secretKey, r.config.SessionToken, r.logger)
```
Comment (fairclothjm) [inline redis_elasticache_client.go:52]: Why are we setting the SessionToken when previously we were not?

Comment (maxcoulombe) [reply redis_elasticache_client.go:52]: It lets users use STS credentials and it seems to be a common AWS authentication option we front on AWS plugins that require an access key id & secret access key, like the [Vault Agent].(https://developer.hashicorp.com/vault/docs/agent/autoauth/methods/aws)

Temporary creds might be ill-suited for a plugin though? Maybe we better not make it available unless a user asks for it and can provide a legitimate use case where managing STS creds is necessary.

**Review by fairclothjm — APPROVED** [review]

File: redis_elasticache_client.go (hunk 3 above)
Comment (fairclothjm) [reply redis_elasticache_client.go]: I agree with that. Thanks!

**Pull request merged**